code/datums/abilities/critter/mindeater.dm

```diff
@@ -0,0 +1,390 @@
+		src.updateText()
+
+ABSTRACT_TYPE(/datum/targetable/critter/mindeater)
+/datum/targetable/critter/mindeater
+	icon = 'icons/mob/critter/nonhuman/intruder.dmi'
+	icon_state = "template"
+	/// reveals the mindeater on use
+	var/reveals_on_use = FALSE
+
+	cast(atom/target)
+		..()
+		if (src.reveals_on_use)
+			var/mob/living/critter/mindeater/mindeater = src.holder.owner
+			mindeater.reveal()
+
+	proc/get_nearest_human_or_silicon(atom/target)
+		if (isliving(target))
+			return target
+		for (var/mob/living/L in view(1, get_turf(target)))
+			if (!(ishuman(L) || issilicon(L)))
+				continue
+			return L
+
+	proc/get_nearest_living(atom/target)
+		if (isliving(target))
+			return target
+		for (var/mob/living/L in view(1, get_turf(target)))
+			if (!(ishuman(L) || issilicon(L) || (iscritter(L) && !istype(L, /mob/living/critter/mindeater))))
+				continue
+			return L
+
+/datum/targetable/critter/mindeater/become_tangible
+	name = "Manifest"
+	desc = "Merge yourself into reality, becoming tangible."
+	icon_state = "manifest"
+	cooldown = 60 SECONDS
+
+	cast(atom/target)
+		. = ..()
+		var/mob/living/critter/mindeater/mindeater = src.holder.owner
+		mindeater.manifest()
+
+/datum/targetable/critter/mindeater/regenerate
+	name = "Regenerate"
+	desc = "Consume Brain to regenerate health."
+	icon_state = "regenerate"
+	pointCost = 5
+
+	tryCast()
+		var/mob/living/critter/mindeater/mindeater = src.holder.owner
+		if (mindeater.get_health_percentage() >= 1)
+			boutput(mindeater, SPAN_ALERT("You're already at full health!"))
+			return CAST_ATTEMPT_FAIL_NO_COOLDOWN
+		return ..()
+
+	cast(atom/target)
+		. = ..()
+		if (actions.hasAction(src.holder.owner, /datum/action/bar/private/mindeater_regenerate))
+			actions.stop(/datum/action/bar/private/mindeater_regenerate, src.holder.owner)
+		else
+			actions.start(new /datum/action/bar/private/mindeater_regenerate(), src.holder.owner)
+
+/datum/targetable/critter/mindeater/brain_drain
+	name = "Brain Drain"
+	desc = "Drain 3 brain per second from a target in range."
+	icon_state = "brain_drain"
+	targeted = TRUE
+	target_anything = TRUE
+	max_range = 6
+
+	tryCast(atom/target)
+		target = src.get_nearest_human_or_silicon(target)
+		var/mob/living/L = target
+		if (!(istype(L, /mob/living/carbon/human) || istype(L, /mob/living/silicon)))
+			boutput(src.holder.owner, SPAN_ALERT("You can only target humans and silicons!"))
+			return CAST_ATTEMPT_FAIL_NO_COOLDOWN
+		if (istype(L, /mob/living/carbon/human))
+			if (L.get_brain_damage() > INTRUDER_MAX_BRAIN_THRESHOLD)
+				boutput(src.holder.owner, SPAN_ALERT("This target has received too much brain damage!"))
+				return CAST_ATTEMPT_FAIL_NO_COOLDOWN
+		if (isdead(L))
+			boutput(src.holder.owner, SPAN_ALERT("You can only use this ability on alive targets!"))
+			return CAST_ATTEMPT_FAIL_NO_COOLDOWN
+		return ..()
+
+	cast(atom/target)
+		. = ..()
+		actions.start(new /datum/action/bar/private/mindeater_brain_drain(target), src.holder.owner)
```

Comment:
This doesn't require line of sight to be maintained and gives no feedback to the victim, meaning it seems to be possible to 100-0 brain drain someone while fully invisible and hiding behind a wall.

Reply:
I was planning to add more feedback in the next stage of mindeaters if all goes well with the initial playtest, I'll add some more "temporary" feedback rn